R/figures.R

```diff
@@ -0,0 +1,220 @@
+#' Get plot of fit and forecasted counts
+#'
+#' @param draws A dataframe containing the posterior draws with the data joined
+#' to it. This is `draws_df` output of the call to `wwinference()`
+#' @param count_data_eval A dataframe containing the count data we will
+#' evaluate the forecasts against.
+#' @param forecast_date A string indicating the date we made the forecast, for
+#' plotting, in ISO8601 format YYYY-MM-DD
+#' @param count_type A string indicating what data the counts refer to,
+#' default is `hospital admissions`
+#'
+#' @return A ggplot object containing the posterior draw of the estimated,
+#' nowcasted, and forecasted hospital admissions alongside the data used to
+#' calibrate the hospital admissions model and the later observed hospital
+#' admissions used to evaluate the forecast performance
+#' @export
+#'
+get_plot_forecasted_counts <- function(draws,
+                                       count_data_eval,
+                                       forecast_date,
+                                       count_type = "hospital admissions") {
+  p <- ggplot(draws |> dplyr::filter(
+    name == "pred_counts",
+    draw %in% sampled_draws
+  )) +
+    geom_line(aes(x = date, y = pred_value, group = draw),
+      color = "red4", alpha = 0.1, size = 0.2
+    ) +
+    geom_point(
+      data = count_data_eval,
+      aes(x = date, y = daily_hosp_admits_for_eval),
```

Comment:
@kaitejohnson

Reply:
Yeah definitely -- I think the better way to do it would be to keep the name for the example data (since the example is for hospital admissions) but pass it as an argument to this fitting function.